src/script/builder/tir/op.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef TVM_SCRIPT_BUILDER_TIR_OP_H_
+#define TVM_SCRIPT_BUILDER_TIR_OP_H_
+
+#include <tvm/tir/expr.h>
+#include <tvm/tir/op.h>
+
+#include "../builder.h"
+
+namespace tvm {
+namespace script {
+namespace builder {
+namespace tir {
+
+static const PrimType int8 = PrimType(DataType::Int(8));
+static const PrimType int16 = PrimType(DataType::Int(16));
+static const PrimType int32 = PrimType(DataType::Int(32));
+static const PrimType int64 = PrimType(DataType::Int(64));
+
+static const PrimType uint8 = PrimType(DataType::UInt(8));
+static const PrimType uint16 = PrimType(DataType::UInt(16));
+static const PrimType uint32 = PrimType(DataType::UInt(32));
+static const PrimType uint64 = PrimType(DataType::UInt(64));
+
+static const PrimType float8 = PrimType(DataType::Float(8));
+static const PrimType float16 = PrimType(DataType::Float(16));
+static const PrimType float32 = PrimType(DataType::Float(32));
+static const PrimType float64 = PrimType(DataType::Float(64));
+
+static const PrimType bool_ = PrimType(DataType::Bool());
```

Comment:
This is probably not the case. In fact, in most of the cases, `T.int8` means type casting, i.e. a method that does:

```cc
PrimExpr int8(PrimExpr e) {
  return tvm::cast(DataType::Int(8), e);
}
```